packages/contracts-sdk/src/lib/contracts-sdk.ts

```diff
@@ -1616,36 +1633,29 @@ https://developer.litprotocol.com/v3/sdk/wallets/auth-methods/#auth-method-scope
           throw new Error('Could not get mint cost');
         }
 
-        let sentTx;
-
         if (this.isPKP) {
           this.log(
             "This is a PKP wallet, so we'll use the PKP wallet to sign the tx"
           );
+        }
+
+        this.log('...populating tx');
 
-          this.log('...populating tx');
+        let _gasLimit: BigNumberish | undefined = param?.gasLimit;
+        if (!_gasLimit) {
           const tx =
             await this.pkpNftContract.write.populateTransaction.mintNext(2, {
               value: mintCost,
-              gasLimit: param?.gasLimit || GAS_LIMIT,
             });
-          this.log('tx:', tx);
-
-          this.log('...signing tx');
-          const signedTx = await this.signer.signTransaction(tx);
```

Comment:
Is the new code functionally still the same after removing the explicit code here that calls an explicit `signTransaction` and `sendTransaction()` using `this.signer` and skip straight to calling `mintNext()` without signing it explicitly using `this.signer`?

Reply:
Not that it is a new code, it is unifying both paths. Now that there might be a intermediate step, the code ended up being identical for both so there was no more need for the `if`

Reply:
I'm just confused in that I don't see this call to `signTransaction` anywhere in the code path anymore; it seems like now we always run the code from the `!this.isPKP` branch instead?

Reply:
`!isPKP` path was populating, signing and sending the transaction directly in one step (line 1643)
`isPKP` path was populating the transaction (line 1627) then signing (line 1635) and sending (line 1639) it with the embedded signer

With the changes for this tkt, `!isPKP` path had to be separated into different steps, becoming almost identical (using the contract interfaces instead of the embedded signer/provider)

But the contracts are initialized with the embedded `signer` and `provider` [here](https://github.com/LIT-Protocol/js-sdk/blob/7327abc9b4cefec1e4bfa2fbbc6efdcaf1c87b77/packages/contracts-sdk/src/lib/contracts-sdk.ts#L484) so both paths really became identical just with different references

The only thing that changes when `isPKP === true` is [the provider](https://github.com/LIT-Protocol/js-sdk/blob/7327abc9b4cefec1e4bfa2fbbc6efdcaf1c87b77/packages/contracts-sdk/src/lib/contracts-sdk.ts#L391) and is then passed to contracts anyway